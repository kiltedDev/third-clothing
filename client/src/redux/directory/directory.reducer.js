// diverged from course here because in a RW scenario we'd be importing
// from a database and I'd like to have this structured more in that
// direction.  p sure they'll do that later anyhow.

// import DIRECTORY_DATA from '../../components/directory/directory.data';

const INITIAL_STATE = {
  sections: []
};

const directoryReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    default:
    return state;
  }
}

export default directoryReducer;
