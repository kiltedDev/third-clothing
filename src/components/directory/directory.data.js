const DIRECTORY_DATA = [
  {
    title: 'hats',
    imageUrl: 'https://i.ibb.co/cvpntL1/hats.png',
    linkUrl: 'hats',
    id: 1
  },
  {
    title: 'jackets',
    imageUrl: 'https://i.ibb.co/px2tCc3/jackets.png',
    linkUrl: 'jackets',
    id: 2
  },
  {
    title: 'sneakers',
    imageUrl: 'https://i.ibb.co/0jqHpnp/sneakers.png',
    linkUrl: 'sneakers',
    id: 3
  },
  {
    title: 'womens',
    imageUrl: 'https://i.ibb.co/GCCdy8t/womens.png',
    size: 'large',
    linkUrl: 'womens',
    id: 4
  },
  {
    title: 'mens',
    imageUrl: 'https://i.ibb.co/R70vBrQ/men.png',
    size: 'large',
    linkUrl: 'mens',
    id: 5
  }
]

export default DIRECTORY_DATA;
