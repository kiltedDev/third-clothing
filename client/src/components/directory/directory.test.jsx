import { shallow } from 'enzyme';
import React from 'react';
import { Directory } from './directory.component';

describe('Directory', () => {
  let wrapper;

  beforeAll(() => {
    const mockProps = {
      sections: [{
        id: 1
      }]
    }
    wrapper = shallow(<Directory { ...mockProps } />)
  })
  it('should render Directory component',() => {
    expect( wrapper.debug() ).toMatchSnapshot();
  })
})
