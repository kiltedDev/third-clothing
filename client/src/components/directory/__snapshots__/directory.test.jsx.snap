// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory should render Directory component 1`] = `
"<div className=\\"directory-menu\\">
  <withRouter(MenuItem) />
</div>"
`;
